tests/Database/SelectFields/RelationshipsTest/MorphRelationshipTest.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rebing\GraphQL\Tests\Database\SelectFields\RelationshipsTest;
+
+use Rebing\GraphQL\Tests\TestCaseDatabase;
+use Rebing\GraphQL\Tests\Support\Models\Like;
+use Rebing\GraphQL\Tests\Support\Models\Post;
+use Rebing\GraphQL\Tests\Support\Models\User;
+use Rebing\GraphQL\Tests\Support\Traits\SqlAssertionTrait;
+
+class MorphRelationshipTest extends TestCaseDatabase
+{
+    use SqlAssertionTrait;
+
+    public function testMorphRelationship(): void
+    {
+        $user = factory(User::class)->create([
+            'name' => 'User Name',
+        ]);
+
+        $otherUser = factory(User::class)->create();
+
+        $post = factory(Post::class)->create([
+            'user_id' => $user->id,
+        ]);
+
+        $like = factory(Like::class)->make([
+            'user_id' => $otherUser->id,
+        ]);
+
+        $post->likes()->save($like);
+
+        $this->assertNotNull($user->posts[0]->likes[0]->user);
+
+        $query = <<<'GRAQPHQL'
+{
+  users {
+    id
+    posts {
+      id
+      title
+      likes {
+        id
+        user {
+            id
+        }
+      }
+    }
+  }
+}
+GRAQPHQL;
+
+        $this->sqlCounterReset();
+
+        $result = $this->graphql($query, [
+            'expectErrors' => true,
+        ]);
+
+        $this->assertSqlQueries(
+            <<<'SQL'
+select "users"."id" from "users";
+select "posts"."id", "posts"."title", "posts"."user_id" from "posts" where "posts"."user_id" in (?, ?) order by "posts"."id" asc;
+select "likes"."id", "likes"."user_id", "likes"."likable_id", "likes"."likable_type" from "likes" where "likes"."likable_id" in (?) and "likes"."likable_type" = ?;
```

Comment:
I feel stupid to ask, but when I leave the tests but rollback the changes in `SelectFields`, this test here only fails due to the different SQL statement (`"likes"."likable_type"` is not included).

But the result assertions are exactly the same. So from the outside, it's not observable.

Can you elaborate?

Reply:
I refactored the test to add test assertions inside the resolve method, to better reflect what it is fixing

Reply:
Oh wow, totally missed that.

But ... it's also _not_ failing for the master branch version of `SelectFields`. What am I missing?

Reply:
I forgot to remove the expectError thingy